src/operation_manager.rs

```diff
@@ -0,0 +1,530 @@
+                    let op_name = self
+                        .selected_ops
+                        .last()
+                        .expect("Failed to get last selected operation")
+                        .op
+                        .name
+                        .clone();
+
+                    // We care only about results with var idents larger than
+                    // the geometry stack length. This means that only new
+                    // geometries are going to be processed.
+                    let last_used_var_ident = self.geometry_stack.len();
+                    let mut new_geometry_metadata = HashMap::new();
+                    let mut var_idents = HashSet::new();
+
+                    for (var_ident, value) in &value_set.used_values {
+                        if var_ident.0 + 1 > last_used_var_ident as u64 {
+                            let geometry_metadata = GeometryMetadata {
+                                name: format!("Geometry #{} from {}", 1, op_name),
+                                geometry: value.unwrap_geometry().clone(),
+                                var_ident: var_ident.0,
+                                used: true,
+                                geometry_id: None,
+                            };
+                            new_geometry_metadata.insert(var_ident.0, geometry_metadata);
+                            var_idents.insert(var_ident.0);
+                        }
+                    }
+
+                    for (var_ident, value) in value_set.unused_values {
+                        if var_ident.0 + 1 > last_used_var_ident as u64 {
+                            let geometry_metadata = GeometryMetadata {
+                                name: format!("Geometry #{} from {}", 1, op_name),
+                                geometry: value.unwrap_geometry().clone(),
+                                var_ident: var_ident.0,
+                                used: false,
+                                geometry_id: None,
+                            };
+                            new_geometry_metadata.insert(var_ident.0, geometry_metadata);
+                            var_idents.insert(var_ident.0);
+                        }
+                    }
+
+                    // Since ordering matters in case of stacks, var idents need
+                    // to be sorted.
+                    let mut var_idents: Vec<_> = var_idents.iter().collect();
+                    var_idents.sort();
+
+                    // Geometry metadata are pushed onto geometry stacks in order.
+                    // Geometry handler runs for each unused geometry so it can
+                    // be added to scene.
+                    for var_ident in var_idents {
+                        let mut geometry_metadata = new_geometry_metadata
+                            .remove(&var_ident)
+                            .expect("Failed to remove new geometry");
+
+                        if !geometry_metadata.used {
```

Comment:
Well, there's 2 options here:

- We will render just the unused geometry, in which case we don't need to keep track of used geometry at all,
- We will render used geometry differently than unused, in which case we will also want to add it to the renderer.